docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+include::{es-test-dir}/cluster/config/analysis/synonym.txt[]
+--------------------------------------------------
+
+To update an existing synonyms set, upload new files to your cluster.
+Synonyms set files must be kept in sync on every cluster node.
+
+When a synonyms set is updated, search analyzers that use it need to be refreshed using the <<indices-reload-analyzers,reload search analyzers API>>
+
+This manual syncing and reloading makes this approach less flexible than using the <<synonyms-store-synonyms-api,synonyms API>>.
+
+[discrete]
+[[synonyms-store-synonyms-inline]]
+==== Inline
+
+For testing out your synonyms, you can directly inline them in your token filter definition.
+
+[WARNING]
+======
+This is not recommended for production usage.
+A large number of inline synonyms increases cluster size unnecessarily and can lead to performance issues.
+======
+
+[discrete]
+[[synonyms-synonym-analyzers]]
+=== Configure your analyzers to use synonyms sets
+
+[discrete]
+[[synonyms-synonym-token-filters]]
+==== Configure synonyms token filters
+
+Once your synonyms sets are created, you can start configuring your token filters and analyzers to use them.
+
+{es} uses synonyms as part of the <<analysis-overview,analysis process>>.
+There are two different <<analysis-tokenfilters,token filters>> that can be used for including synonyms:
+
+* <<analysis-synonym-tokenfilter>>: Use it for single word synonyms ("car", "automobile").
+* <<analysis-synonym-graph-tokenfilter>>: Use it for synonyms that may span multiple words ("hurriedly", "in a hurry").
+
+Synonyms sets created via synonyms APIs can be included into synonym token filters using `synonyms_set` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms_set": "my-synonym-set"
+    }
+  }
+----
+
+The synonym file path needs to be included as part of your token filter using `synonyms_path` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms_path": "analysis/synonym-set.txt"
+    }
+  }
+----
+
+Synonyms can be directly included in your token filter using the `synonyms` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms": ["pc => personal computer", "computer, pc, laptop"]
+    }
+  }
+----
+
+[discrete]
+[[synonyms-analizers-configure]]
+==== Configure analyzers with synonym token filters
+
+To apply synonyms, you will need to include one of the above synonym token filters into an analyzer, like the following:
+
+[source,JSON]
+----
+      "analyzer": {
+        "my_analyzer": {
+          "type": "custom",
+          "tokenizer": "standard",
+          "filter": ["lowercase", "synonyms_filter"]
+        }
+      }
+----
+
+Order is important for your token filters.
+{es} will use the token filters preceding the synonym filter to parse the synonym rules in a synonyms set.
```

Comment:
parse => pass

Reply:
I copied that directly from the synonyms graph filter, and it didn't sound right. Thanks for catching this!